bitnami/mongodb/values.yaml

```diff
@@ -175,10 +175,30 @@ tls:
   ## @param tls.autoGenerated Generate a custom CA and self-signed certificates
   ##
   autoGenerated: true
-  ## @param tls.existingSecret Existing secret with TLS certificates (keys: `mongodb-ca-cert`, `mongodb-ca-key`, `client-pem`)
-  ## NOTE: When it's set it will disable certificate creation
+  ## @param tls.existingSecret Existing secret with TLS certificates (keys: `mongodb-ca-cert`, `mongodb-ca-key`)
+  ## NOTE: When it's set it will disable secret creation.
   ##
   existingSecret: ""
+  standalone:
+    ## @param tls.standalone.existingSecret Existing secret with TLS certificates (`tls.key`, `tls.crt`, `ca.crt`).
+    ## NOTE: When it's set it will disable certificate self-generation from existing CA.
+    ##
+    existingSecret: ""
+  replicaset:
+    ## @param tls.replicaset.existingSecrets Array of existing secrets with TLS certificates (`tls.key`, `tls.crt`, `ca.crt`).
+    ## NOTE: When it's set it will disable certificate self-generation from existing CA.
+    ##
+    existingSecrets: []
+  hidden:
+    ## @param tls.hidden.existingSecrets Array of existing secrets with TLS certificates (`tls.key`, `tls.crt`, `ca.crt`).
+    ## NOTE: When it's set it will disable certificate self-generation from existing CA.
+    ##
+    existingSecrets: []
+  arbiter:
+    ## @param tls.arbiter.existingSecret Existing secret with TLS certificates (`tls.key`, `tls.crt`, `ca.crt`).
+    ## NOTE: When it's set it will disable certificate self-generation from existing CA.
+    ##
+    existingSecret: ""
```

Comment:
This is one possible approach, but I believe it's kind of complex as it adds a lot of entries under the `tls` param (especially, considering that we already have entries for `hidden` and `arbiter`). 

This is because the MongoDB chart is complex itself, it has a lot of components and different replication configurations. Taking a look into Kafka (which is also a complex chart), we can see that TLS for its components was implemented under their specific entry in the `values.yaml` file (e.g. `provisioning.auth.tls`).

https://github.com/bitnami/charts/blob/1d59ac4394f68177dd349a2008675448bc82f081/bitnami/kafka/values.yaml#L1478-L1487

I think we should follow the same approach here and add `tls` entries for the different components. Moreover, we already follow this schema for other params such as _Service_:

https://github.com/bitnami/charts/blob/1d59ac4394f68177dd349a2008675448bc82f081/bitnami/mongodb/values.yaml#L1795-L1797

----

In that case of the main MongoDB pods, we have to bear in mind that we have two architectures: `standalone` and `replicaset`. 

I would simply use a single array param, `existingSecrets`, and consider `standalone` as a special case of `replicaset` where only a single entry is expected. WDYT?

Reply:
Hi @joancafom, thank you very much for your feedback! I agree with you, it adds a lot of entry and I already considered a lot of cases when doing these changes. Let me add some more thoughts. At first, I added a single `existingSecrets` containing all secrets for all pods. Fewer parameters, but definitively too complex to manage and use.   
I then considered two options. The one you explained, and the one I implemented. I think each option has some advantages. When all `tls` parameters are in the same entries, it is easier to see what is required to configure tls. The elasticsearch chart is using this method: https://github.com/bitnami/charts/tree/master/bitnami/elasticsearch. All elasticsearch components (master, data, coordinating) required to run an elasticsearch cluster have an entry `security.tls.*component*`. On the other hand, having a specific `*component*.tls` can be interesting as well. In the case of the Kafka chart, it only needs brokers to run a Kafka cluster. The Kafka provisioning is just acting as a temporary client to configure the Kafka brokers. So it can make sense that the Kafka provisioning has its own `tls` entry.  
That says, in the case of MongoDB, to configure autogenerated TLS in all MongoDB components (hidden, replica, arbiter), you have to configure TLS in the main `tls` entry only. Nothing inside `hidden` and `arbiter` entries. So if we put the `tls` entry in `hidden` and `arbiter` entries, there are now two different cases and it may be confusing:
- Autogenerated CA or CA provided+autogenerated certs -> only `tls` entry
- Custom certs -> some entries in `tls` entry, some in `hidden.tls` entry, and some in `arbiter.tls` entry.  

That's why I finally choose to implement the `tls.*component*` entries solution. To my mind, it is not necessarily more complex, but definitively less confusing in this case. However, sharing a single secret array for both `standalone` and `replicaset` may be a good idea to reduce the number of parameters and the complexity :+1: Not sure how to do that to avoid confusion though.  What do you think? :)

Reply:
Okay, I see your points now and I personally like that we already have this implementation in another chart :)

Apologies for the delay, I will add this PR to my reviewing list and provide further feedback 😁

Reply:
Hi @joancafom, thank you! I'm waiting for your feedback then :D